Sources/BBCodeParser.php

```diff
@@ -2742,7 +2742,7 @@ protected function parseMessage(): void
 				$data = strtr($data, ["\t" => '&nbsp;&nbsp;&nbsp;']);
 
 				// If it wasn't changed, no copying or other boring stuff has to happen!
-				if ($data != substr($this->message, $this->last_pos, $this->pos - $this->last_pos)) {
+				if ($data != substr($this->message, $this->last_pos, (int) ($this->pos - $this->last_pos))) {
```

Comment:
This is unnecessary. The arithmetic operation will automatically cast the resulting value to int anyway.